src/rrules/avoiding_non_differentiable_code.jl

```diff
@@ -88,6 +88,9 @@ import Base.CoreLogging as CoreLogging
     }
 )
 
+# zero_rdata_from_type must handle new object types.
+zero_rdata_from_type(::Type{T}) where {T<:Base.CoreLogging.AbstractLogger} = NoRData()
```

Comment:
I suggest we add a unit test for this, so it is easier to debug in the future.

Reply:
> We already have this.

By which I mean that if `rdata_type(tangent_type(P))` is `NoRData`, `zero_rdata_from_type` will just return `NoRData()`.